document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/mapping/smallfull/mappers/ApiToPeriodMapperDecorator.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.companieshouse.document.generator.accounts.mapping.smallfull.mappers;
+
+import uk.gov.companieshouse.accountsdates.AccountsDatesHelper;
+import uk.gov.companieshouse.accountsdates.impl.AccountsDatesHelperImpl;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.model.company.account.LastAccountsApi;
+import uk.gov.companieshouse.api.model.company.account.NextAccountsApi;
+import uk.gov.companieshouse.document.generator.accounts.mapping.smallfull.model.ixbrl.period.Period;
+
+import java.time.LocalDate;
+
+
+public abstract class ApiToPeriodMapperDecorator implements ApiToPeriodMapper {
+
+    private final ApiToPeriodMapper apiToPeriodMapper;
+
+    private AccountsDatesHelper accountsDatesHelper = new AccountsDatesHelperImpl();
+
+    public ApiToPeriodMapperDecorator(ApiToPeriodMapper apiToPeriodMapper) {
+        this.apiToPeriodMapper = apiToPeriodMapper;
+    }
+
+    @Override
+    public Period apiToPeriod(CompanyProfileApi companyProfile) {
+
+        Period period = apiToPeriodMapper.apiToPeriod(companyProfile);
+
+        boolean isSameYear = isSameYearFiler(companyProfile);
+
+        NextAccountsApi nextAccounts = companyProfile.getAccounts().getNextAccounts();
+        period.setCurrentPeriodStartOnFormatted(convertToDisplayDate(nextAccounts.getPeriodStartOn()));
+        period.setCurrentPeriodEndOnFormatted(convertToDisplayDate(nextAccounts.getPeriodEndOn()));
+        period.setCurrentPeriodBSDate(convertDateToBSDate(nextAccounts.getPeriodStartOn(),
+                nextAccounts.getPeriodEndOn(), isSameYear));
+
+
+        if (isMultipleYearFiler(companyProfile)) {
+            LastAccountsApi lastAccounts = companyProfile.getAccounts().getLastAccounts();
```

Comment:
we could move this to its own method `setPreviousPeriod(period, companyProfile)` same with the current period